module/VuFind/src/VuFind/Controller/AlmaController.php

```diff
@@ -0,0 +1,342 @@
+<?php
+/**
+ * Alma controller
+ *
+ * PHP version 5
+ *
+ * Copyright (C) AK Bibliothek Wien für Sozialwissenschaften 2018.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Michael Birkner <michael.birkner@akwien.at>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:controllers Wiki
+ */
+namespace VuFind\Controller;
+
+use \Zend\ServiceManager\ServiceLocatorInterface;
+use \Zend\Http\Response as HttpResponse;
+//use \Zend\Http\Request as HttpRequest;
+//use \Zend\Mail as Mail;
+
+
+class AlmaController extends AbstractBase {
+    
+    /**
+     * Http service
+     * @var \VuFindHttp\HttpService
+     */
+    protected $httpService;
+    
+    /**
+     * Http response
+     * @var \Zend\Http\Response
+     */
+    protected $httpResponse;
+    
+    /**
+     * Http headers
+     * @var \Zend\Http\Headers
+     */
+    protected $httpHeaders;
+    
+    /**
+     * config.ini config
+     * @var \Zend\Config\Config
+     */
+    protected $config;
+    
+    /**
+     * Alma.ini config
+     * @var \Zend\Config\Config
+     */
+    protected $configAlma;
+    
+    /**
+     * User table
+     * 
+     * @var \VuFind\Db\Table\User
+     */
+    protected $userTable;    
+        
+    /**
+     * Alma Controler constructor.
+     * 
+     * @param ServiceLocatorInterface $sm
+     */
+    public function __construct(ServiceLocatorInterface $sm) {
+    	
+        parent::__construct($sm);
+        $this->httpResponse = new HttpResponse();
```

Comment:
Is there a specific reason you are creating your own response object here? The framework already provides a getResponse() method that will give you the same type of object. Perhaps you can simplify slightly to take advantage of that.